root/static/scripts/guess-case/utils.js

```diff
@@ -87,6 +87,14 @@ var lowerCaseBracketWordsList = [
 
 var lowerCaseBracketWords = new RegExp('^(' + lowerCaseBracketWordsList.join('|') + ')$', 'i');
 
+exports.turkishUpperCase = function(str) {
+  return str.normalize().replace(/i/g, 'İ').toUpperCase();
+};
+
+exports.turkishLowerCase = function(str) {
+  return str.normalize().replace(/I/g, 'ı').replace(/İ/g, 'i').toLowerCase();
+};
```

Comment:
Hmm. I guess this is a matter of taste; but NFC normalization is a rather heavy action that may change a number of unrelated things, too. (We will do it on the server later anyway.) I would prefer a simple `.replace(/I\u0307/g, 'i').replace(/İ/g, 'i')`.

And for the uppercasing function, normalization doesn’t serve a purpose.